src/metabase/driver/sql_jdbc/sync/describe_table.clj

```diff
@@ -0,0 +1,164 @@
+(ns metabase.driver.sql-jdbc.sync.describe-table
+  "SQL JDBC impl for `describe-table` and `describe-table-fks`."
+  (:require [clojure.java.jdbc :as jdbc]
+            [clojure.string :as str]
+            [clojure.tools.logging :as log]
+            [medley.core :as m]
+            [metabase.driver :as driver]
+            [metabase.driver.sql-jdbc.connection :as sql-jdbc.conn]
+            [metabase.driver.sql-jdbc.sync
+             [common :as common]
+             [interface :as i]])
+  (:import [java.sql Connection DatabaseMetaData]))
+
+(defmethod i/column->special-type :sql-jdbc [_ _ _] nil)
+
+(defn pattern-based-database-type->base-type
+  "Return a `database-type->base-type` function that matches types based on a sequence of pattern / base-type pairs."
+  [pattern->type]
+  (fn [column-type]
+    (let [column-type (name column-type)]
```

Comment:
why `loop` instead of something like `m/find-first`?

Reply:
I haven't changed this at all but that's a good change IMO